### PR TITLE
PM-5017 Migrate Organization Plans component

### PR DESCRIPTION
**Pull request by KiruthigaManivannan**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [x] Other
```

## Objective

Migrate the Organization Plans component to use the component library.

## Code changes

- **organization-plans.component.html** Changed Radio button groups, checkboxes, Form input fields , buttons and all heading and paragraph tags to its equivalent components in component library.
- **organization-plans.component.ts** Added Form group for file import and removed try/catch in submit click.

## Screenshots
- **Click on Upgrade plan button click:**
Before:
image
image
image

After:
image
image
image

### CODE REVIEW

File: apps/web/src/app/billing/organizations/organization-plans.component.html [hunk 1]
```diff
@@ -1,400 +1,436 @@
...
+      (changedBusinessOwned)="changedOwnedBusiness()"
+      [formGroup]="formGroup"
+      [createOrganization]="createOrganization"
+      [isProvider]="!!providerId"
+      [acceptingSponsorship]="acceptingSponsorship"
+    >
+    </app-org-info>
+  </bit-section>
+  <bit-section>
+    <h2 bitTypography="h2" class="tw-mt-5">{{ "chooseYourPlan" | i18n }}</h2>
```
Comment (willmartian) [inline apps/web/src/app/billing/organizations/organization-plans.component.html:53]: Section titles shouldn't need extra margin/padding

File: apps/web/src/app/billing/organizations/organization-plans.component.html [hunk 2]
```diff
@@ -1,400 +1,436 @@
...
+    <div *ngFor="let selectableProduct of selectableProducts" class="">
```
Comment (willmartian) [inline apps/web/src/app/billing/organizations/organization-plans.component.html:54]: Remove empty attribute `class`

File: apps/web/src/app/billing/organizations/organization-plans.component.html [hunk 3]
```diff
@@ -1,400 +1,436 @@
...
+  <bit-section>
+    <div *ngIf="formGroup.value.product !== productTypes.Free">
+      <bit-section
         *ngIf="
-          !selectableProduct.PasswordManager.basePrice &&
-          selectableProduct.PasswordManager.hasAdditionalSeatsOption
+          selectedPlan.PasswordManager.hasAdditionalSeatsOption &&
+          !selectedPlan.PasswordManager.baseSeats
         "
       >
```
Comment (willmartian) [inline apps/web/src/app/billing/organizations/organization-plans.component.html:201]: Is it necessary to have the following hierarchy?

```html
<bit-section>
  <div>
    <bit-section>
      ...
    </bit-section>
    ...
  </div>
</bit-section>
```

Seems like we should just be able to do:

```html
<bit-section>...</bit-section>
<bit-section>...</bit-section>
...
```

File: apps/web/src/app/billing/organizations/organization-plans.component.html [hunk 4]
```diff
@@ -1,400 +1,436 @@
...
+          <bit-form-field class="tw-col-span-6">
+            <bit-label>{{ "userSeats" | i18n }}</bit-label>
+            <input
+              bitInput
+              type="number"
+              formControlName="additionalSeats"
+              placeholder="{{ 'userSeatsDesc' | i18n }}"
+              required
+            />
+            <bit-hint bitTypography="body2">{{ "userSeatsHowManyDesc" | i18n }}</bit-hint>
```
Comment (willmartian) [inline apps/web/src/app/billing/organizations/organization-plans.component.html:213]: `bitTypography` does not replace the need to add `tw-text-muted`

File: apps/web/src/app/billing/organizations/organization-plans.component.html [hunk 5]
```diff
@@ -1,400 +1,436 @@
...
-            type="checkbox"
-            name="premiumAccessAddon"
-            formControlName="premiumAccessAddon"
-          />
-          <label for="premiumAccess" class="form-check-label bold">{{
-            "premiumAccess" | i18n
-          }}</label>
+      </bit-section>
+      <bit-section>
+        <h2 bitTypography="h2" class="tw-mt-5">{{ "addons" | i18n }}</h2>
```
Comment (willmartian) [inline apps/web/src/app/billing/organizations/organization-plans.component.html:218]: Don't add additional margin/padding to headers

File: apps/web/src/app/billing/organizations/organization-plans.component.html [hunk 6]
```diff
@@ -1,400 +1,436 @@
...
+        <div
+          class="tw-grid tw-grid-cols-12"
```
Comment (willmartian) [inline apps/web/src/app/billing/organizations/organization-plans.component.html:220]: Missing grid gap

File: apps/web/src/app/billing/organizations/organization-plans.component.html [hunk 7]
```diff
@@ -1,400 +1,436 @@
...
+        >
+          <bit-form-field class="tw-col-span-6">
+            <bit-label>{{ "additionalUserSeats" | i18n }}</bit-label>
+            <input
+              bitInput
+              type="number"
+              formControlName="additionalSeats"
+              placeholder="{{ 'userSeatsDesc' | i18n }}"
+            />
+            <bit-hint bitTypography="body2"
```
Comment (willmartian) [inline apps/web/src/app/billing/organizations/organization-plans.component.html:234]: Don't add custom typography styles to CL components such as `bit-hint`

File: apps/web/src/app/billing/organizations/organization-plans.component.html [hunk 8]
```diff
@@ -1,400 +1,436 @@
...
-            {{ selectablePlan.PasswordManager.basePrice | currency: "$" }}
-            {{ "monthAbbr" | i18n }}
-            =
-            {{ selectablePlan.PasswordManager.basePrice | currency: "$" }}
-            /{{ "month" | i18n }}
-          </small>
-          <small *ngIf="selectablePlan.PasswordManager.hasAdditionalSeatsOption">
-            <span *ngIf="selectablePlan.PasswordManager.baseSeats"
-              >{{ "additionalUsers" | i18n }}:</span
+        <div class="tw-grid tw-grid-cols-12">
```
Comment (willmartian) [inline apps/web/src/app/billing/organizations/organization-plans.component.html:244]: Missing grid gap

File: apps/web/src/app/billing/organizations/organization-plans.component.html [hunk 9]
```diff
@@ -1,400 +1,436 @@
...
+                  : "1 GB"
+                  : (additionalStoragePriceMonthly(selectedPlan) | currency: "$")
+                  : ("month" | i18n)
+            }}</bit-hint>
+          </bit-form-field>
+        </div>
+      </bit-section>
+      <bit-section>
+        <div
+          class="tw-grid tw-grid-cols-12"
```
Comment (willmartian) [inline apps/web/src/app/billing/organizations/organization-plans.component.html:266]: Missing grid gap

File: apps/web/src/app/billing/organizations/organization-plans.component.html [hunk 10]
```diff
@@ -1,400 +1,436 @@
...
+          <bit-form-field class="tw-col-span-6">
+            <bit-label>{{ "additionalStorageGb" | i18n }}</bit-label>
+            <input
+              bitInput
+              type="number"
+              formControlName="additionalStorage"
+              step="1"
+              placeholder="{{ 'additionalStorageGbDesc' | i18n }}"
+            />
+            <bit-hint bitTypography="body2">{{
```
Comment (willmartian) [inline apps/web/src/app/billing/organizations/organization-plans.component.html:254]: Don't add custom typography styles to CL components such as `bit-hint`

File: apps/web/src/app/billing/organizations/organization-plans.component.html [hunk 11]
```diff
@@ -1,400 +1,436 @@
...
+            <input type="checkbox" bitCheckbox formControlName="premiumAccessAddon" />
+            <bit-hint bitTypography="body2">{{
+              "premiumAccessDesc" | i18n: (3.33 | currency: "$") : ("month" | i18n)
+            }}</bit-hint>
+          </bit-form-control>
+        </div>
+      </bit-section>
+      <bit-section>
+        <h2 bitTypography="h2">{{ "summary" | i18n }}</h2>
+        <div *ngFor="let selectablePlan of selectablePlans">
```
Comment (willmartian) [inline apps/web/src/app/billing/organizations/organization-plans.component.html:280]: Is this div needed?

File: apps/web/src/app/billing/organizations/organization-plans.component.html [hunk 12]
```diff
@@ -1,400 +1,436 @@
...
+              <bit-label>{{ (selectablePlan.isAnnual ? "annually" : "monthly") | i18n }}</bit-label>
+              <bit-hint *ngIf="selectablePlan.isAnnual">
+                <p
+                  class="tw-mb-0"
+                  bitTypography="body2"
+                  *ngIf="selectablePlan.PasswordManager.basePrice"
+                >
+                  {{ "basePrice" | i18n }}:
+                  {{
+                    (selectablePlan.isAnnual
```
Comment (willmartian) [inline apps/web/src/app/billing/organizations/organization-plans.component.html:296]: Replace inline style (`style="text-decoration: line-through"`) in this block with Tailwind

File: apps/web/src/app/billing/organizations/organization-plans.component.html [hunk 13]
```diff
@@ -1,400 +1,436 @@
...
       [selectedPlan]="selectedSecretsManagerPlan"
       [upgradeOrganization]="!createOrganization"
     ></sm-subscribe>
-  </div>
+  </bit-section>
 
   <!-- Payment info -->
   <div *ngIf="formGroup.value.product !== productTypes.Free">
-    <h2 class="mb-4">
+    <h2 bitTypography="h2" class="tw-mb-4">
```
Comment (willmartian) [inline apps/web/src/app/billing/organizations/organization-plans.component.html:421]: Use `bit-section` instead of div + margin

**Review by willmartian — CHANGES_REQUESTED** [review]

**Review by amorask-bitwarden — APPROVED** [review]

**Comment by amorask-bitwarden** [comment]
@vinith-kovan This branch did not build when you merged it. Can you please ensure the build checks pass before you merge the migration PRs?

**Pull request merged**